scipy/stats/_fit.py

```diff
@@ -913,6 +914,18 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
     larger number of them exceed the value of statistic for `data`, thereby
     inflating the p-value.
 
+    *Implementation Notes*
+
+    The Filliben statistic is calculated using the theoretical medians of the
```

Comment:
[Here](https://www.jstor.org/stable/1268008) is the paper. Can you access it?

I think this part is relevant. Filliben has just noted that 
![image](https://user-images.githubusercontent.com/6570539/224582668-3c3f868f-a184-4b0c-b8e5-b80d80a98734.png)

He states that closed-form expressions for the $m_i$ do not exist, but provides this approximation:

![image](https://user-images.githubusercontent.com/6570539/224582731-de64f9af-9b98-4222-bd21-140bdf6c4b26.png)

I didn't use this approximation in the Filliben PR. [The order statistics of the uniform distribution follow a beta distribution](https://en.wikipedia.org/wiki/Order_statistic#Order_statistics_sampled_from_a_uniform_distribution), and SciPy can calculate the median of a beta distribution, so that's what I used in the Filliben PR. 

I wrote "theoretical" here to avoid writing "exact", because the $m_i$ probably cannot, in general, be represented _exactly_ in floating point arithmetic.